tools/unix/apply-big-move.sh

```diff
@@ -0,0 +1,181 @@
+#! /bin/bash
+#
+# This script applies the most recent "big change".
+#
+# This script is designed so that it can help you rebase an in-flight PR across this change. Once
+# the big move has been merged, you may simply run this script inside your branch, and it will
+# take care of updating your change in order to avoid any merge conflicts.
+#
+# How it does this:
+# 1. Rebases your change to the commit just before the Big Move.
+# 2. Applies the same logical changes that the Big Move did, but now applies this to the changes
+#    on your branch too.
+# 3. Rewrites git history so that your commits are now based on the commit immediately after the
+#    Big Move.
+#
+# To use the script, just run it in your branch. It'll take care of everything, and you'll end up
+# with the changes in your branch all based on the commit after The Big Move.
+
+set -euo pipefail
+
+AFTER_MOVE_TAG=after-big-move-1
+BEFORE_MOVE_TAG=before-big-move-1
+NEXT_BIG_MOVE=before-big-move-2
+ORIG_BRANCH=$(git branch --show-current)
+
+fail() {
+  echo -e '\e[0;1;31mFAILED:\e[0m '"$@" >&2
+  exit 1
+}
+
+check_big_move_dependencies() {
+  source "$(dirname -- $BASH_SOURCE)/find-python3.sh"
+  PYTHON_PATH=$(get_python3)
+  if [[ -z "$PYTHON_PATH" ]]; then
+    fail "Cannot find python3. Run install python3 first."
+  fi
+}
+
+apply_big_move() {
+  echo "Applying format..."
+
+  source "$(dirname -- $BASH_SOURCE)/find-python3.sh"
+  PYTHON_PATH=$(get_python3)
+  $PYTHON_PATH "$(dirname -- $BASH_SOURCE)/../cross/format.py"
```

Comment:
FWIW this is a slightly different approach taken than the internal big move script in that you are running against only those files changed wrt. the HEAD commit rather than all files which should be formatted.

The intention of the original approach was that you can use this script to actually apply the big move in the first place by running `apply-big-move.sh --apply` so you can be sure you are doing the same (idempotent) thing when applying the big move in the first place and then applying it to each branch's commit.

That being said, assuming you run clang-format on (at least) the set of files that the `format.py` script might format, this should still work fine.

Reply:
Well, the format.py script is the only way I used to clang-format so it should match :)

Reply:
Ohh, I just realized that you actually changed the semantics of the original script when porting it - you added the `--git` option to the script when porting it. This makes a lot more sense now.